### PR TITLE
Base  WMI operations for save and pause.

### PR DESCRIPTION
**Pull request by jonah-kohn**
This PR is one of many to enable save/pause VM operations. See the entire list of related PRs below: 

WMI: Here. 
Moc-pkg: https://github.com/microsoft/moc-pkg/pull/466
Wssdagent: https://github.com/microsoft/wssdagent/pull/885
Wssdcloudagent: https://github.com/microsoft/wssdcloudagent/pull/898
moc: https://github.com/microsoft/moc/pull/218
moc-sdk-for-go: https://github.com/microsoft/moc-sdk-for-go/pull/193

### CODE REVIEW

File: pkg/virtualization/core/virtualsystem/virtualmachine.go [hunk 1]
```diff
@@ -218,6 +218,24 @@ func (vm *VirtualMachine) Start() error {
 	return vm.WaitForState(Running, StateChangeTimeoutSeconds)
 }
 
+// Pause Virtual Machine
+func (vm *VirtualMachine) Pause() error {
+	err := vm.ChangeState(Pausing, v2.ConcreteJob_JobType_Pause_Virtual_Machine, -1)
+	if err != nil {
+		return err
+	}
+	return vm.WaitForState(Pausing, StateChangeTimeoutSeconds)
```
[the remark this answers is not in the document]
Comment (jonah-kohn) [reply pkg/virtualization/core/virtualsystem/virtualmachine.go:227]: I assumed the same, but I followed the precedent set for start and stop operations where we wait for state "starting"

Let me know if I should change all of them to wait for "paused" or "running" etc.

[the remark this answers is not in the document]
Comment (jonah-kohn) [reply pkg/virtualization/core/virtualsystem/virtualmachine.go:227]: Agreed. Updated.

**Review by madhanrm — APPROVED** [review]

**Comment by madhanrm** [comment]
Good to merge

**Comment by jonah-kohn** [comment]
@Madhan re-requesting review after minor revisions

File: pkg/virtualization/core/virtualsystem/virtualmachine.go [hunk 2]
```diff
@@ -218,6 +218,33 @@ func (vm *VirtualMachine) Start() error {
 	return vm.WaitForState(Running, StateChangeTimeoutSeconds)
 }
 
+// Resume Virtual Machine
+func (vm *VirtualMachine) Resume() error {
```
Comment (jonah-kohn) [inline pkg/virtualization/core/virtualsystem/virtualmachine.go:222]: It is technically possible to just "Start" VMs from Paused or Saved state, but this is extremely simple to add and probably better practice.

Comment (msftpgayam) [reply pkg/virtualization/core/virtualsystem/virtualmachine.go:222]: But it could also create confusion. Let's add some comments here and wmi code

**Review by msftpgayam — APPROVED** [review]

**Review by raghavendra-nataraj — DISMISSED** [review]

File: server2019/root/virtualization/v2/ConcreteJob_JobType.go [hunk 3]
```diff
@@ -37,9 +38,9 @@ const (
 	// Shut_Down_Virtual_Machine enum
 	ConcreteJob_JobType_Shut_Down_Virtual_Machine ConcreteJob_JobType = 16
 	// Pause_Virtual_Machine enum
-	ConcreteJob_JobType_Pause_Virtual_Machine ConcreteJob_JobType = 26
+	ConcreteJob_JobType_Pause_Virtual_Machine ConcreteJob_JobType = 20
```
Comment (madhanrm) [inline server2019/root/virtualization/v2/ConcreteJob_JobType.go:41]: You are not supposed to change this file. It would be over written if re-generated

Comment (madhanrm) [reply server2019/root/virtualization/v2/ConcreteJob_JobType.go:41]: If these enums are incorrect, create a local enum in virtualmachine.go and use it.

And Add comments as to why this enum is incorrect

**Review by raghavendra-nataraj — APPROVED** [review]

**Pull request merged**